src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import duke.task.Task;
+import exception.DukeException;
+
+/**
+ * Deals with interaction with the user
+ */
+public class Ui {
+
+
+    /**
+     * Says bye to the user
+     *
+     * @return farewell message
+     */
+    public String bye() {
+        return "Bye. Hope to see you again!";
+    }
+
+    /**
+     * Shows the user the duke error message
+     *
+     * @param e the exception
+     * @return error message
+     */
+    public String showErrorMessage(DukeException e) {
+        return e.getMessage();
+    }
+
+    /**
+     * Informs the user that there are no tasks in the list
+     *
+     * @return no task message
+     */
+    public String showNoTaskMessage() {
+        return "There are no tasks!";
+    }
+
+    /**
+     * Shows the user the current task list
+     *
+     * @param taskList the list of the tasks
+     * @return task list
+     */
+    public String showTaskList(TaskList taskList) {
+        StringBuilder list = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 1; i < taskList.size() + 1; i++) {
+            String s = String.format("  %d.%s%n\n", i, taskList.get(i));
+            list.append(s);
+        }
+        return list.toString();
+    }
+
+    /**
+     * Informs the user that the task is deleted
+     *
+     * @param deletedTask the deleted task
+     * @param taskListSize the number of tasks in list after deletion
+     * @return delete message
+     */
+    public String showDeleteMessage(Task deletedTask, int taskListSize) {
+        return String.format("Noted. I've removed this task:\n  %s\nNow you have %d task(s) in the list.%n",
+                deletedTask, taskListSize);
+    }
+
+    /**
+     * Informs the user that the task is added
+     *
+     * @param newTask the task to add to list
+     * @param taskListSize the number of tasks in list after adding new task
+     * @return add task message
+     */
+    public String showAddTaskMessage(Task newTask, int taskListSize) {
+        return String.format("Got it. I've added this task:\n  %s\nNow you have %d task(s) in the list.%n",
+                newTask, taskListSize);
+    }
+
+    /**
+     * Informs the user that the task is marked as done
+     *
+     * @param doneTask the task after marking as done
+     * @return done message
+     */
+    public String showDoneMessage(Task doneTask) {
+        return "Nice! I've marked this task as done:\n  " + doneTask;
```

Comment:
May i know why on some show*** methods you use String.format and others u use String addition?